src/main/scala/uk/sky/fs2/kafka/topicloader/TopicLoader.scala

```diff
@@ -67,75 +71,85 @@ trait TopicLoader {
   def loadAndRun[F[_] : Async : LoggerFactory, K, V](
       topics: NonEmptyList[String],
       consumerSettings: ConsumerSettings[F, K, V]
-  )(onLoad: Resource.ExitCase => F[Unit]): Stream[F, ConsumerRecord[K, V]] =
-    KafkaConsumer
-      .stream(consumerSettings)
-      .flatMap { consumer =>
-        load(topics, LoadAll, consumer).onFinalizeCase(onLoad) ++ consumer.records.map(_.record)
-      }
+  )(onLoad: Resource.ExitCase => F[Unit]): Stream[F, ConsumerRecord[K, V]] = {
+    given Logger[F] = LoggerFactory[F].getLogger
+
+    def postLoad(logOffsets: NonEmptyMap[TopicPartition, LogOffsets]): Stream[F, ConsumerRecord[K, V]] =
+      for {
+        // The only consistent workaround for re-assigning offsets after the initial load is to re-create the consumer
+        postLoadConsumer <- KafkaConsumer.stream(consumerSettings)
+        _                <- Stream.eval(assignOffsets(logOffsets, postLoadConsumer)(_.highest))
+        record           <- postLoadConsumer.records.map(_.record)
+      } yield record
```

Comment:
This is the main bit of the change. We seemed to have had a race condition where the `load` stream would finish, and if the `onLoad` took too long it would poll the consumer again and increase the next fetch offset to n+1, meaning we would sometimes drop the next message to be read. This surfaced in the tests, as you can see from the previous prbuilds they have a 50/50 chance of passing due to this race condition.

By re-creating the consumer, it stops this race condition by destroying the preload consumer that would poll to the next offset, and re-creates the consumer with the correct offsets.